lib/mix/lib/mix/tasks/profile.cprof.ex

```diff
@@ -0,0 +1,333 @@
+  count traced.
+
+  To profile the code, you can use syntax similar to the `mix run` task:
+
+      mix profile.cprof -e Hello.world
+      mix profile.cprof -e "[1, 2, 3] |> Enum.reverse |> Enum.map(&Integer.to_string/1)"
+      mix profile.cprof my_script.exs arg1 arg2 arg3
+
+  ## Command line options
+
+    * `--func-spec` - a {module, function, arity} expression or 'on_load' to set a trace pattern
+    * `--limit`, - filters out any results with a call count less than the limit
+    * `--module`, - filters out any results not pertaining to the given module
+    * `--config`, `-c`  - loads the given configuration file
+    * `--eval`, `-e` - evaluate the given code
+    * `--require`, `-r` - requires pattern before running the command
+    * `--parallel`, `-p` - makes all requires parallel
+    * `--no-compile` - does not compile even if files require compilation
+    * `--no-deps-check` - does not check dependencies
+    * `--no-archives-check` - does not check archives
+    * `--no-halt` - does not halt the system after running the command
+    * `--no-start` - does not start applications after compilation
+    * `--no-elixir-version-check` - does not check the Elixir version from mix.exs
+
+  ## Profile output
+  
+  Example output:
+                                                                           CNT
+      Total                                                                 15
+      Enum                                                                   6  <--
+        Enum."-map/2-lists^map/1-0-"/2                                       4
+        Enum.reverse/1                                                       1
+        Enum.map/2                                                           1
+      :elixir_compiler                                                       4  <--
+        anonymous fn/1 in :elixir_compiler.__FILE__/1                        3
+        anonymous fn/0 in :elixir_compiler.__FILE__/1                        1
+      String.Chars.Integer                                                   3  <--
+        String.Chars.Integer.to_string/1                                     3
+      :erlang                                                                2  <--
+        :erlang.trace_pattern/3                                              2
+      Profile done over 20229 matching functions
+
+  The default output contains data gathered from all matching functions. The left
+  column structures each module and its total call count trace is presented on the right.
+  Each module has its count discriminated by function below. The `<--` symbol is meant to
+  help visualize where a new module call count begins.
+
+  The first row (Total) is the sum of all function calls. In the last row the number of
+  matching functions that were considered for profiling is presented.
+
+  When `--func-spec` option is specified, call count tracing will be started only for 
```

Comment:
In Elixir we don't use `func` to refer to functions anywhere and I don't think `spec` here is helpful. An alternative name I propose is `--from-mfa`, as we start counting from this particular module+function+arity.